### DISCUSSION TITLE
Postprocessing: compute a residual like quantity with assembly

### DISCUSSION THREAD

**Discussion by rcontinuum** [idaholab/moose#21677] [Q&A Modules: Solid mechanics] [answered]
Hello, 
I'm using the `TotalLagrangianStressDivergence` Kernel with the NeoHookean material model. After each time step I want to compute in a *postprocessing* step a nodal residual *like* quantity

$R^\alpha = \int_V T_{iK}\phi^\alpha_{i,K}  dV$

with $T$ a stress tensor (not the PK1!) on element level and then *assemble* the element vector (a type of element "internal" force vector) to a global vector, e.g. an `[AuxVariable]`. I want to be able to do postprocessing with this global nodal vector and write it to a exodus file. Again: this computation does not contribute to the residual of the system of nonlinear equations. It is just postprocessing.

If I understand correct then, but I'm quite new to MOOSE, `[AuxKernel]` does not give me access to the gradient of the test functions nor to the assembly function. 
Thus, conceptually, I would like to have an `[AuxKernel]` that is like an `[Kernel]` but does assemble into a user-defined nodal  `[AuxVariable]`.

Any idea how to do this? 

From the `Kernel.C` code
https://mooseframework.inl.gov/docs/doxygen/moose/Kernel_8C_source.html
```
90  void
91  Kernel::computeResidual()
92  {
93  prepareVectorTag(_assembly, _var.number());
94 
95  precalculateResidual();
96  for (_i = 0; _i < _test.size(); _i++)
97    for (_qp = 0; _qp < _qrule->n_points(); _qp++)
98      _local_re(_i) += _JxW[_qp] * _coord[_qp] * computeQpResidual();
99 
100  accumulateTaggedLocalResidual();
```
but the assembly should go to a nodal `[AuxVariable]` defined in the input file  and *not* contribute to the global residual.
Additionally, I would have to compute in `computeQpResidual()` or name it  `computeQpAuxResidual()` my stress $T_{iK}$

`Kernel::computeResidual()` is called by `LagrangianStressDivergenceBase` in
https://mooseframework.inl.gov/docs/doxygen/modules/LagrangianStressDivergenceBase_8C_source.html
```
75  void
76  LagrangianStressDivergenceBase::computeResidual()
77  {
78    precalculateResidual();
79    Kernel::computeResidual();
80  }
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

I think the easiest way to do this will be to use the tagging interface
https://mooseframework.inl.gov/framework_development/tagging.html
While computing your kernels, store this quantity in a tagged vector then it can be retrieved by the postprocessors system later

Guillaume

**Reply by rcontinuum** [reply]
Dear Guillaume,

thank you for this hint. I got it running.

Cheers
